language/stdlib/modules/validator_set.mvir

```diff
@@ -74,53 +74,108 @@ module ValidatorSet {
     // TODO: Decide on access control policy. For now, we ensure that this is only callable from the
     // genesis txn. Obviously, we'll need a different policy once we support reconfiguration.
     add_validator(account_address: address) acquires T {
-        let config: ValidatorConfig.Config;
-        let config_ref: &ValidatorConfig.Config;
         let validator_set_ref: &mut Self.T;
-        let vector_ref: &mut Vector.T<Self.ValidatorInfo>;
 
-        // A prospective validator must have an account
-        assert(LibraAccount.exists(copy(account_address)), 17);
-
-        config = ValidatorConfig.config(copy(account_address));
-        config_ref = &config;
+        // A prospective validator must have a validator config resource
+        assert(ValidatorConfig.has(copy(account_address)), 17);
 
         validator_set_ref = borrow_global_mut<T>(0x1D8);
-
         Vector.push_back<Self.ValidatorInfo>(
             &mut move(validator_set_ref).validators,
             ValidatorInfo {
                 addr: move(account_address),
-                consensus_pubkey: ValidatorConfig.consensus_pubkey(copy(config_ref)),
+                consensus_pubkey: h"",
                 // TODO: check for LIT, compute voting power based on LIT + stake
                 consensus_voting_power: 1,
-                network_signing_pubkey: ValidatorConfig.network_signing_pubkey(copy(config_ref)),
-                network_identity_pubkey: ValidatorConfig.network_identity_pubkey(move(config_ref)),
+                network_signing_pubkey: h"",
+                network_identity_pubkey: h"",
             }
         );
 
         return;
     }
 
+   // Return true if the ValidatorInfo given as input is different than the ValidatorConfig
+    // published at validator_info.addr. Aborts if there is no ValidatorConfig at
+    // validator_info.addr
+    public validator_info_changed(validator_info: &mut Self.ValidatorInfo): bool {
```

Comment:
the function name is a little confusing to me, it not only tests the change but also applies it

Reply:
Will change the name.